src/features/metadata-based-view/MetadataQueryAPIHelper.js

```diff
@@ -0,0 +1,140 @@
+        }
+
+        const fields = Object.keys(instance)
+            .filter(key => !key.startsWith('$'))
+            .map(key => {
+                const templateField = find(templateFields, ['key', key]);
+                const type = getProp(templateField, 'type'); // get data type
+                const field: MetadataQueryInstanceTypeField = {
+                    name: key,
+                    value: instance[key],
+                    type,
+                };
+
+                if (includes(SELECT_TYPES, type)) {
+                    // get "options" for enums or multiselects
+                    field.options = getProp(templateField, 'options');
+                }
+
+                return field;
+            });
+
+        return {
+            enterprise: {
+                fields,
+                id: instance.$id,
+            },
+        };
+    };
+
+    flattenResponseEntry = ({ item, metadata }: MetadataQueryResponseEntry): BoxItem => {
+        const { id, name, size } = item;
+
+        return {
+            id,
+            metadata: this.flattenMetadata(metadata),
+            name,
+            size,
+        };
+    };
+
+    getFlattenedDataWithTypes = (templateSchemaResponse?: MetadataTemplateSchemaResponse): Collection => {
+        this.metadataTemplate = getProp(templateSchemaResponse, 'data');
+        const { entries, next_marker }: MetadataQueryResponseData = this.metadataQueryResponseData;
+        return {
+            items: entries.map<BoxItem>(this.flattenResponseEntry),
+            nextMarker: next_marker,
+        };
+    };
+
+    filterMetdataQueryResponse = (response: MetadataQueryResponseData): MetadataQueryResponseData => {
+        const { entries = [], next_marker } = response;
+        return {
+            entries: entries.filter(entry => getProp(entry, 'item.type') === ITEM_TYPE_FILE), // return only file items
+            next_marker,
+        };
+    };
+
+    getTemplateSchemaInfo = (data: MetadataQueryResponseData): Promise<MetadataTemplateSchemaResponse | void> => {
+        const { entries } = data;
+        this.metadataQueryResponseData = this.filterMetdataQueryResponse(data);
+        if (!entries || entries.length === 0) {
+            // Don't make metadata API call to get template info
+            return Promise.resolve();
+        }
+
+        const metadata = getProp(entries, '[0].metadata');
+        this.templateScope = Object.keys(metadata)[0];
+        const instance = metadata[this.templateScope];
+        this.templateKey = Object.keys(instance)[0];
+
+        return this.api.getMetadataAPI(true).getSchemaByTemplateKey(this.templateKey);
+    };
+
+    queryMetadata = (metadataQuery: MetadataQueryType): Promise<MetadataQueryResponseData> => {
+        return new Promise((resolve, reject) => {
+            this.api.getMetadataQueryAPI().queryMetadata(metadataQuery, resolve, reject, { forceFetch: true });
+        });
+    };
+
+    fetchMetadataQueryResults = (
```

Comment:
You can return the promise here to avoid passing in both callbacks. Is this a fetch call?

Reply:
Yes. Keeping it consistent with ContentExplorer usage of callbacks. Its a fetch call to metadata and metadata_queries api endpoints.